security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -226,6 +230,25 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 	return ns, nil
 }
 
+// SecretExist checks if secret already existed.
```

Comment:
The comment does not provide too much info. Suggest to change to something that describes why we need this function.